src/dirwatcher/dirwatcher.js

```diff
@@ -0,0 +1,47 @@
+import fs from 'fs';
+import { EventEmitter } from 'events';
+
+export class DirWatcher extends EventEmitter{
+    watch(path, delay){
+        let files = [];
+        let latestFiles = [];
+        setInterval(()=>{
+            latestFiles = fs.readdirSync(path);
+            if(Array.isArray(files) && files.length){
+                // For addition of files
+                let change = this.compare(files, latestFiles);
+                if(change) {
+                    this.emit('changed', change);
+                }else{
+                    //For changes in any files.
+                    files.forEach((file, index) => {
+                        if((Date.now() - fs.statSync(path+'/'+file).mtimeMs) < 1000){
+                            this.emit('changed',latestFiles[index]);
+                        }
+                    });
+                }
+            }
+            files = fs.readdirSync(path);
+        }, delay);
+    }
+
+    compare(filesArray, latestFilesArray){
+        let element = '';
+        if(filesArray.length === latestFilesArray.length){
+            for(let i=0; i<filesArray.length; i++) {
+                if(latestFilesArray[i] != filesArray[i]){
+                    element = latestFilesArray[i];
+                }
+            };
+            return element;
+        }
+        else{
+            for(let i=0; i<Math.max(filesArray.length, latestFilesArray.length); i++) {
+                if(latestFilesArray[i] !== filesArray[i]){
+                    element = filesArray[i]?filesArray[i]:latestFilesArray[i];
+                }
+            };
+            return element;
```

Comment:
In both `if` and `else` you're `return element`, please extract it out of conditions so you'll have
```
if () {} else {}
return element;
```

Reply:
Thanks for pointing out. Corrected